stores/bench_test.go

```diff
@@ -114,6 +114,52 @@ func BenchmarkPruneHostSectors(b *testing.B) {
 	}
 }
 
+type noopUpdater struct{}
+
+func (n noopUpdater) UpdateElementProof(*types.StateElement) {
+	// do nothing
+}
+
+// BenchmarkUpdateWalletProofElements benchmarks updating wallet proof elements.
+//
+// 254135490 ns/op | M1 Max | SQLite
+// 254135490 ns/op | M1 Max | MySQL
```

Comment:
When you say 15x considering that this is 0.25s per batch, it was almost 4s per batch before? 

Another potential tweak could be pulling updating the proofs out of `revertChainUpdate` and `applyChainUpdate`.
Instead, we can fetch all the elements at the beginning of `processUpdates` I think, only update their proofs in memory for each `cru` and `cau` and then persist the update at the end of `processUpdates`.
So when we process a batch of 100 updates we only perform 1 read+insert rather than 100.

It would be interesting to see if that already improves the testing situation since it's a potentially simpler change which doesn't require updating the query.

Reply:
Yes, on my machine on MySQL it's 4429677209 ns/op. 

Pretty good idea to do that in memory and pull that out. I tried it in `renterd` and the interface is slightly awkward because you have to expose a map from the store to keep track of the `db_contract_id`, but it works! It would resolve the testing situation were it not that we are only fixing contract element updates, not wallet elements. We use the wallet implementation from `coreutils` so we would have to change it there as well. That's not really worth it imo, it does make things slightly more awkward.

Reply:
I think it's worth making this change. If you agree, gives this message a thumbs up and I'll split out the impl. for MySQL and SQLite and update the benchmark and push the PR in review.

Reply:
Hm looking at the implementation in `coreutils` I think it might actually not be that easy for the wallet as it is for the contracts.

Because we'd need to account for new elements being created or deleted in `applyChainUpdate` and `revertChainUpdate`. So keeping them cached in memory becomes a bit of a nightmare. It also doesn't play nice with our interface which means we might update `coreutils`, `renterd` and even `hostd`.

Reply:
Building a query like that seems a bit too risky. But considering that a proper fix is a lot more involved I'd settle for changing the test that is slow. e.g. to avoid mining 400 blocks we could manually form the contract for a short duration that allows the test to run a lot faster.

Reply:
Yeah changing it to 5 or so in testing should also work considering we never have more than 5 hosts anyway.